jib-core/src/main/java/com/google/cloud/tools/jib/builder/BuildConfiguration.java

```diff
@@ -210,13 +209,10 @@ public BuildConfiguration build() {
       if (targetImageReference == null) {
         errorMessages.add("target image is required but not set");
       }
-      if (mainClass == null) {
-        errorMessages.add("main class is required but not set");
-      }
 
       switch (errorMessages.size()) {
         case 0: // No errors
-          if (baseImageReference == null || targetImageReference == null || mainClass == null) {
+          if (baseImageReference == null || targetImageReference == null) {
```

Comment:
Should we error if `entrypoint` is empty like we used to error when `mainClass` was null?

Reply:
I'm intending it to be possibly be empty to allow for propagation of entrypoint from base images in the future (when this is used as a library).